### PR TITLE
prov/verbs: ep_rdm bugfix

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
- Fix truncated error handling in rndv protocol (tagged send/recv)
 - Fix inconsistent max_inline check in tagged inject
 - Remove types with variable size from fi_ibv_rdm_buf and fi_ibv_rdm_buf_service_data structures
 - Other minor refactoring

@a-ilango @shefty

### CODE REVIEW

File: prov/verbs/src/ep_rdm/verbs_utils.h [hunk 1]
```diff
@@ -115,14 +115,15 @@ do {                                                                        \
     const size_t max_str_len = 1024;                                        \
     char str[max_str_len];                                                  \
     snprintf(str, max_str_len,                                              \
-            "%s request: %p, eager_state: %s, rndv_state: %s, tag: 0x%lx, len: %lu, rest: %lu, context: %p, connection: %p\n", \
+            "%s request: %p, eager_state: %s, rndv_state: %s, err_state: %ld, tag: 0x%lx, len: %lu, rest: %lu, context: %p, connection: %p\n", \
```
Comment (shefty) [inline prov/verbs/src/ep_rdm/verbs_utils.h]: This is an impressively long line.

**Comment by evgeny-leksikov** [comment]
However, this is very helpful in debugging. It is excluded from release build.

**Comment by shefty** [comment]
I was merely suggesting splitting across multiple lines.

**Comment by evgeny-leksikov** [comment]
It's not a problem for me but coding style guide say: "never break user-visible strings such as
print messages, because that breaks the ability to grep for them"

**Comment by shefty** [comment]
Well, it's awfully hard for the user to grep for "%s request: %p, eager_state: %s, rndv_state: %s"...  What the user sees and what grep can find are vastly different.  :)

**Comment by evgeny-leksikov** [comment]
ok, I'll fix it  :)

**Comment by evgeny-leksikov** [comment]
Fixed + minor refactoring of alignments and naming

**Comment by evgeny-leksikov** [comment]
I'm validating another portion of bufixes, do not merge it before the update

**Comment by evgeny-leksikov** [comment]
It's ready for merge now

**Pull request merged**